packages/@aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-listener.ts

```diff
@@ -48,13 +73,29 @@ export class NetworkListener extends BaseListener implements INetworkListener {
    */
   private readonly loadBalancer: INetworkLoadBalancer;
 
+  /**
+   * Protocol assigned to listener
+   */
+  private readonly protocol: Protocol;
+
+  /**
+   * Certificates array
+   */
+  private readonly certificates?: INetworkListenerCertificateProps[];
+
   constructor(scope: cdk.Construct, id: string, props: NetworkListenerProps) {
     super(scope, id, {
       loadBalancerArn: props.loadBalancer.loadBalancerArn,
-      protocol: Protocol.Tcp,
+      protocol: props.protocol || Protocol.Tcp,
```

Comment:
I think it might make sense to default to `Protocol.Tls` if we detect certificates. A case-sensitive default.

Reply:
Sounds good!